pkg/testutils/e2e/deploy/deploy.go

```diff
@@ -40,21 +41,53 @@ func RunPod(k8s kubernetes.Kubernetes, ns string, pod corev1.Pod, timeout time.D
 	return nil
 }
 
-// ExposeApp deploys a pod in a namespace and exposes it as a service in a cluster
+// RunPodSet runs a set of replicas of an Pod.
+func RunPodSet(k8s kubernetes.Kubernetes, ns string, pod corev1.Pod, replicas int, timeout time.Duration) error {
+	wg := sync.WaitGroup{}
+	errCh := make(chan error, replicas)
+
+	// use the name as a prefix
+	for i := 0; i < replicas; i++ {
+		wg.Add(1)
+
+		go func(pod corev1.Pod, replica int) {
+			defer wg.Done()
+
+			// generate an unique name for the pod
+			pod.Name = fmt.Sprintf("%s-%d", pod.Name, replica)
+
+			if err := RunPod(k8s, ns, pod, timeout); err != nil {
+				errCh <- err
+			}
+		}(pod, i)
+	}
+
+	wg.Wait()
+
+	select {
+	case err := <-errCh:
+		return err
+	default:
+		return nil
+	}
+}
+
+// ExposeApp deploys a number of replicas of a pod in a namespace and exposes them as a service in a cluster
 // The ingress routes request that specify the service's name as host to this service.
 func ExposeApp(
 	k8s kubernetes.Kubernetes,
 	namespace string,
 	pod corev1.Pod,
+	replicas int,
```

Comment:
Should we return an error if `replicas == 0`? Seems like it could be the source of one of annoying to debug situations where nothing happens but no error is returned.